common/src/main/java/com/wynntils/features/tooltips/ItemCompareFeature.java

```diff
@@ -4,162 +4,648 @@
+
+    @Persisted
+    public final Config<Boolean> centerItemName = new Config<>(false);
+
     @RegisterKeyBind
-    private final KeyBind toggleCompareModeKeyBind =
-            new KeyBind("Compare mode", GLFW.GLFW_KEY_KP_EQUAL, true, null, this::onCompareModeKeyPress);
+    private final KeyBind holdToCompareKeyBind =
+            new KeyBind("Hold to compare", GLFW.GLFW_KEY_KP_ENTER, false, null, null);
 
     @RegisterKeyBind
-    private final KeyBind compareSelectKeyBind =
-            new KeyBind("Select for comparing", GLFW.GLFW_KEY_C, true, null, this::onSelectKeyPress);
+    private final KeyBind selectCompareKeyBind =
+            new KeyBind("Select for comparing", GLFW.GLFW_KEY_KP_ADD, true, null, this::onSelectKeyPress);
 
-    private ItemStack comparedItem = null;
-    private boolean compareToEquipped = false;
+    private final List<Pair<WynnItem, ItemStack>> selectedItems = new ArrayList<>();
+    private static final int COMPARE_ITEM_PAD = 6;
+    private static final String EQUIPPED_KEY = "feature.wynntils.itemCompare.tag.equipped";
+    private static final String HOVERED_KEY = "feature.wynntils.itemCompare.tag.hovered";
+    private static final String SELECTED_KEY = "feature.wynntils.itemCompare.tag.selected";
+    private static final String HOVERED_SELECTED_KEY = "feature.wynntils.itemCompare.tag.hovered_selected";
+
+    // First equippedCount items in itemsToCompare will have "Equipped" tag, others will have "Selected" tag
+    private int equippedCount = 0;
+    private boolean changePositioner = false;
+
+    @SubscribeEvent
+    public void onWorldStateChangeEvent(WorldStateEvent event) {
+        selectedItems.clear();
+    }
 
     @SubscribeEvent
-    public void onWorldStateChange(WorldStateEvent event) {
-        comparedItem = null;
-        compareToEquipped = false;
+    public void onContainerCloseEvent(ContainerCloseEvent.Post event) {
+        selectedItems.clear();
     }
 
     @SubscribeEvent
-    public void onRenderSlot(SlotRenderEvent.Pre event) {
+    public void onInventoryClickEvent(ContainerClickEvent event) {
+        unselectItemStack(event.getItemStack());
+    }
+
+    @SubscribeEvent(priority = EventPriority.HIGHEST)
+    public void onTooltipRenderEvent(TooltipRenderEvent event) {
+        if (!changePositioner) return;
+
+        event.setPositioner(PassiveTooltipPositioner.INSTANCE);
+    }
+
+    @SubscribeEvent
+    public void onSlotRenderEvent(SlotRenderEvent.Pre event) {
         Slot slot = event.getSlot();
-        if (slot.getItem() == comparedItem) {
-            RenderUtils.drawArc(event.getPoseStack(), CommonColors.LIGHT_BLUE, slot.x, slot.y, 200, 1, 6, 8);
-        }
+        drawSelectionArc(event.getPoseStack(), slot.getItem(), slot.x, slot.y, false);
     }
 
     @SubscribeEvent
-    public void onItemTooltipRenderEvent(ItemTooltipRenderEvent.Post event) {
+    public void onHotbarSlotRenderEvent(HotbarSlotRenderEvent.Pre event) {
```

Comment:
Is there a reason the selection arc needs to be drawn on the hotbar? Afaik this is for the hotbar in your hud not in the inventory and you clear selected items on closing containers

Reply:
For consistency only tbh. Since hotbar is still visible while inventory is open

Reply:
I suppose it can be kept, I just don't see any reason to display that it's been selected twice when the focus will be on the inventory